datalake/src/main/java/com/sequenceiq/datalake/service/upgrade/OrderedOSUpgradeRequestProvider.java

```diff
@@ -35,23 +43,26 @@ public OrderedOSUpgradeSetRequest createMediumDutyOrderedOSUpgradeSetRequest(Sta
         Map<String, List<String>> instanceIdsByInstanceGroup = getInstanceIdsByInstanceGroup(instanceMetaDataByInstanceGroup);
         LOGGER.debug("Instance ids by instance group: {}", instanceIdsByInstanceGroup);
 
+        int order = 0;
         List<OrderedOSUpgradeSet> osUpgradeByUpgradeSets = new ArrayList<>();
-        osUpgradeByUpgradeSets.add(new OrderedOSUpgradeSet(0, Set.of(
+        osUpgradeByUpgradeSets.add(new OrderedOSUpgradeSet(order++, Set.of(
                 getInstanceId(instanceIdsByInstanceGroup, MASTER),
```

Comment:
It is not your code, but can we rename it to pollInstanceId to express that it removes an element?

Reply:
I renamed order to osUpgradeByUpgradeSetIndex.